### PR TITLE
2867 - Add support for expandable row to expand all frozen columns with datagrid

### LINKED ISSUE #2867: Datagrid: Problem using rowTemplate/expandableRows and frozenColumns with both right and left side frozen.
**Describe the bug**
When using rowTemplate/expandableRows and frozenColumns with both right and left side frozen. The right side is not rendered properly

**To Reproduce**
1. Take the example code for the example http://latest-enterprise.demo.design.infor.com/components/datagrid/test-expandable-row-frozen-columns.html
1. In the code change the frozen columns as 
`frozenColumns: { left: ['productId'], right:['quantity10'] },`
1. Compile and click on expand button and see the right frozen section, the expanded area is missing
1. See Error in the sohoxi.js file

**Expected behavior**
Right table should have a expanded row. so that three sections are aligned and no error.

1. Also as a Second Example Use the same example above, remove rowTemplate setting, add `expandableRow: true` and `onExpandRow: this.expandRowCallback`, and keep both left and right frozen columns. In this case also, the right table expander row is missing.

**Expected result:**
Right table also should have expanded row. Missing the condition in the sohoxi.js file 46797. Need to add the condition.

**Version**
Soho version 4.21

**Screenshots**

**Platform**
Both chrome and IE

### PR DESCRIPTION
**Pull request by deep7102**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Added support for expandable row to expand all frozen columns with Datagrid.

**Related github/jira issue (required)**:
Fixes #2867

**Steps necessary to review your pull request (required)**:
- Pull this branch and build/run the demo app
- Navigate to http://localhost:4000/components/datagrid/test-expandable-row-frozen-columns-bothsides.html
- Click on `+` button in row-1
- Row should expands to all the columns, there should not be any left or right empty column
- http://localhost:4000/components/datagrid/test-expandable-row-frozen-columns-bothsides.html
- Use same steps to test

**Included in this Pull Request**:
- [x] A note to the change log.

### CODE REVIEW

**Comment by ericangeles** [comment]
Looks great to me. Just need to fix the failing test in functional datagrid.

**Comment by tmcconechy** [comment]
Maybe `expandableRowSpanFrozenColumns` ? bit long but tough to name. I like the idea tho.

**Comment by deep7102** [comment]
Hi @tmcconechy updated now,
- name change as `expandRowAcrossAllCells`
- can not reproduce the the point you see as above. 
`when expanding and collapsing one of the columns gets less visible`

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 ### v4.26.0 Features
 
+- `[Datagrid]` Added support for expandable row to expand all frozen columns. ([#2867](https://github.com/infor-design/enterprise/issues/2867))
```
Comment (tmcconechy) [inline docs/CHANGELOG.md]:
```suggestion
- `[Datagrid]` Added support for expandable row to expand across all frozen columns, and fixed span layout issues on the right side frozen columns. ([#2867](https://github.com/infor-design/enterprise/issues/2867))
```

File: src/components/datagrid/datagrid.js [hunk 2]
```diff
@@ -43,6 +43,7 @@ const COMPONENT_NAME = 'datagrid';
  * @param {boolean}  [settings.alternateRowShading=false] Sets shading for readonly grids
  * @param {array}    [settings.columns=[]] An array of columns (see column options)
  * @param {array}    [settings.frozenColumns={ left: [], right: [] }] An object with two arrays of column id's. One for freezing columns to the left side, and one for freezing columns to the right side.
+ * @param {boolean}  [settings.frozenColumns.expandRowAcrossAllCells=true] expand the expandable row to all columns
```
Comment (tmcconechy) [inline src/components/datagrid/datagrid.js]:
```suggestion
 * @param {boolean}  [settings.frozenColumns.expandRowAcrossAllCells=true] Expand the expandable across all frozen columns.
```

**Pull request merged**